core/src/kmx/kmx_plus.cpp

```diff
@@ -131,27 +131,40 @@ header_from_bytes(const uint8_t *data, KMX_DWORD length, uint32_t ident) {
  * @tparam T
  * @param data
  * @param length
+ * @param was_valid on exit, set to false if section was not valid
  * @return const T*
  */
 template <class T>
-const T *section_from_bytes(const uint8_t *data, KMX_DWORD length) {
+const T *section_from_bytes(const uint8_t *data, KMX_DWORD length, bool &was_valid) {
   if (length < sizeof(T)) { // Does not include dynamic data. First check.
     DebugLog("length < sizeof(section)");
     assert(false);
     return nullptr;
   }
   const COMP_KMXPLUS_HEADER *header = header_from_bytes(data, length, T::IDENT);
   const T *section = reinterpret_cast<const T *>(header);
-  if (section != nullptr && section->valid(length)) {
-    return section;
+  if (section != nullptr) {
+    if (!section->valid(length)) {
+      was_valid = false;
+      return nullptr;
+    } else {
+      // valid!
+      return section;
+    }
   } else {
+    // not found
     assert(false);
     return nullptr;
   }
 }
 
+/**
+ * @param sect this is the pointer to the 'sect' section table. Should not be null!
+ * @param was_valid on exit, this is set to false if an invalid section was found. otherwise not modified
```

Comment:
Is this a good pattern? I would have thought that it'd be cleaner to set `was_valid` to `true` if the section is valid. (And call it `is_valid`?) Otherwise, effectively we are asking this function to pass state through that parameter, which is fragile.

Reply:
> That would move from an invalid to a valid state potentially.

Hence why I suggested shifting the concern outside of this function, as this function cares only about the validity of its own section, right? I see you have done that :grin: